### PR TITLE
Add pretransform and optimize build

### PR DESCRIPTION
**Pull request by jhajduk-sage**

### CODE REVIEW

File: dist/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     },
     "description": "Design tokens for the Sage Design System.",
     "author": "The Sage Group plc",
-    "version": "1.86.0",
+    "version": "0.2.0",
```
Comment (JoannaMikul) [inline dist/package.json:9]: this shouldn't be overwritten

File: scripts/utils/ensure-px-units.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+const isNumber = require('lodash/isNumber')
+const isString = require('lodash/isString')
+
+/**
+ * Ensures that input has pixel units if needed
+ * @param input
+ * @returns {string|*}
+ */
+module.exports = (input) => {
+  if (isString(input) && input.includes('px')) {
```
Comment (JoannaMikul) [inline scripts/utils/ensure-px-units.js:13]: maybe one condition is enough ?
if (isString(input) && (input.includes('px') || input === '0' )) { ...

File: yarn.lock [hunk 3]
```diff
@@ -4125,7 +4132,7 @@ tiny-relative-date@^1.3.0:
   resolved "https://registry.npmjs.org/tiny-relative-date/-/tiny-relative-date-1.3.0.tgz"
   integrity sha512-MOQHpzllWxDCHHaDno30hhLfbouoYlOI8YlMNtvKe1zXbjEVhbcEovQxvZrPvtiYW630GQDoMMarCnjfyfHA+A==
 
-tinycolor2@^1.4.1:
+tinycolor2@^1.4.1, tinycolor2@^1.4.2:
```
Comment (JoannaMikul) [inline yarn.lock:4135]: tinycolor2@^1.4.1 ? probably some kind of residue ?

File: scripts/transforms/custom-value-css-box-shadow.transform.js [hunk 4]
```diff
@@ -1,35 +1,32 @@
...
+        return undefined
+      }
 
-    if (isArray(value)) {
-      return value.map(shadowDefinition => {
-        const boxShadow = `${shadowDefinition.x}px ${shadowDefinition.y}px ${shadowDefinition.blur}px ${shadowDefinition.spread}px ${shadowDefinition.color}`
-        if (shadowDefinition.type === 'innerShadow') {
-          return `inset ${boxShadow}`
-        }
+      const boxShadow = `${ensurePxUnits(definition.x)} ${ensurePxUnits(definition.y)} ${ensurePxUnits(definition.blur)} ${ensurePxUnits(definition.spread)} ${definition.color}`
```
Comment (JoannaMikul) [inline scripts/transforms/custom-value-css-box-shadow.transform.js:23]: good improvement ;)

File: package.json [hunk 5]
```diff
@@ -10,7 +10,7 @@
     "url": "https://github.com/Sage/design-tokens"
   },
   "engines": {
-    "node": ">=14"
+    "node": ">=12"
```
Comment (JoannaMikul) [inline package.json:13]: Why did you change the node version?

File: dist/package.json (hunk 1 above)
Comment (jhajduk-sage) [reply dist/package.json:9]: right ;)

File: scripts/utils/ensure-px-units.js (hunk 2 above)
Comment (jhajduk-sage) [reply scripts/utils/ensure-px-units.js:13]: It is split on purpose. I've done it for better code readability.
After all I don't know if one solution is better than the other one

File: scripts/transforms/custom-value-css-box-shadow.transform.js (hunk 4 above)
Comment (jhajduk-sage) [reply scripts/transforms/custom-value-css-box-shadow.transform.js:23]: thanks! :D

File: yarn.lock (hunk 3 above)
Comment (jhajduk-sage) [reply yarn.lock:4135]: style-dictionary uses the `1.4.1` version:
```
style-dictionary@^3.0.1:
    [...]
    tinycolor2 "^1.4.1"
```
and omod and design-tokens library are using `1.4.2` version.

File: dist/package.json (hunk 1 above)
Comment (jhajduk-sage) [reply dist/package.json:9]: Actually whole `/dist` folder shouldn't be commited.

File: package.json (hunk 5 above)
Comment (jhajduk-sage) [reply package.json:13]: I needed to run the library locally on node 12. I needed node12 common js module support. I'll recheck again if everything will be working on node 14.

**Review by JoannaMikul — APPROVED** [review]
